src/content/blog/astro-480.mdx

```diff
@@ -0,0 +1,193 @@
+
+To use this feature, first enable the `experimental.actions` option in your Astro config:
+
+```js
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+	experimental: {
+		actions: true
+	}
+})
+```
+
+Then, define actions in the `src/actions/index.ts` file using the `defineAction` function from `astro:actions`. You can define actions that accept JSON or form requests, and the `handler` function will be called with the parsed input and the context object:
+
+```ts
+// src/actions/index.ts
+import { defineAction, z } from "astro:actions";
+
+export const server = {
+  like: defineAction({
+    // accept json
+    input: z.object({ postId: z.string() }),
+    handler: async ({ postId }, context) => {
+      // ... update likes in db
+      return likes;
+    },
+  }),
+  comment: defineAction({
+    // accept form requests
+    accept: "form",
+    input: z.object({
+      postId: z.string(),
+      author: z.string(),
+      body: z.string(),
+    }),
+    handler: async ({ postId }, context) => {
+      // ... insert comments in db
+      return comment;
+    },
+  }),
+};
+```
+
+You can then, call the actions from your client components using the `actions` object from `astro:actions`. You can pass a type-safe object when using JSON, or a `FormData` object when using `accept: 'form'` in your action definition:
+
+```jsx
+// src/components/blog.tsx
+import { actions } from "astro:actions";
+import { useState } from "preact/hooks";
+
+export function Like({ postId }: { postId: string }) {
+  const [likes, setLikes] = useState(0);
+  return (
+    <button
+      onClick={async () => {
+        const newLikes = await actions.like({ postId });
+        setLikes(newLikes);
+      }}
+    >
+      {likes} likes
+    </button>
+  );
+}
+
+export function Comment({ postId }: { postId: string }) {
+  return (
+    <form
+      onSubmit={async (e) => {
+        e.preventDefault();
+        const formData = new FormData(e.target);
+        const result = await actions.blog.comment(formData);
+        // handle result
+      }}
+    >
+      <input type="hidden" name="postId" value={postId} />
+      <label for="author">Author</label>
+      <input id="author" type="text" name="author" />
+      <textarea rows={10} name="body"></textarea>
+      <button type="submit">Post</button>
+    </form>
+  );
+}
+```
+
+For more information on Astro actions, visit [TODO: link to docs].
+
+## Experimental: Rewriting support
+
+Astro 4.8 introduces experimental support for rewriting. Rerouting allows you to show the content of a different page from the current URL. This can be useful for showing different content based on the user's location, device, or other conditions, without needing multiple different URLs. This feature is sometimes also called "rerouting" in other frameworks.
```

Comment:
I do think having *some* other word with rewriting *feels* better, whatever is correct. (and optionally a way to sneak "routing" in there?) But I know Matthew researched and most frameworks do call this feature rewriting, so it may not be necessary.

```suggestion
Astro 4.8 introduces experimental support for Request rewriting. Rewriting is a routing feature that allows you to show the content of a different page from the current URL. This can be useful for showing different content based on the user's location, device, or other conditions without needing multiple different URLs. This feature is sometimes also called "rerouting" in other frameworks.
```